### DISCUSSION TITLE
Install for GitHub Actions?

### DISCUSSION THREAD

**Discussion by jeremyhaile** [pgpartman/pg_partman#644] [General]
We use GitHub Actions for CI, and I'm having trouble figuring out any commands which can install pg_partman. Has anyone had any success installing the pg_partman extension as part of a GitHub Actions workflow?

**Comment by jeremyhaile** [comment]
OK I'm getting closer. When I build pg_partman I'm seeing this:
> /bin/mkdir -p '/usr/share/postgresql/14/extension'

But then when I run my database scripts I get this error:
> Could not open extension control file "/usr/share/postgresql/15/extension/pg_partman.control"

So it looks like it's installing into a PG 14 directory even though I'm using PG 15. Is there an option I can pass to `make install` to specify which folder I want to use?

**Comment by jeremyhaile** [comment]
Ah I see that I am using a services container...so I need to figure out how to run the commands in the services container. Or...find a container that already has pg_partman installed.

**Comment by jeremyhaile** [comment]
I was able to get this working by creating a custom docker file that runs the commands to install pg_partman.

In case this is helpful to anyone else, here is our Dockerfile for PG 15:

```
FROM postgres:15-bullseye

RUN apt-get update

RUN apt-get install -y git sudo make

RUN git clone https://github.com/pgpartman/pg_partman.git &&\
    cd pg_partman &&\
    sudo make NO_BGW=1 install

RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

**Comment by keithf4** [comment]
Thank you for the inquiry and also providing the solution you found!
